src/libopensc/asn1.c

```diff
@@ -706,17 +713,28 @@ static int encode_bit_field(const u8 *inbuf, size_t inlen,
 	return encode_bit_string(data, bits, outbuf, outlen, 1);
 }
 
-int sc_asn1_decode_integer(const u8 * inbuf, size_t inlen, int *out)
+int sc_asn1_decode_integer(const u8 * inbuf, size_t inlen, int *out, int strict)
```

Comment:
isn't it technically more of "non_compliant" rather than non "strict"?

DER is DER, the decoding and encoding of an object needs to always produce the same bytes...

Reply:
Yes. But it is already used in other places for very similar encoding, sometimes prefixed with zero bytes, or on fixed length bytes, ...). I think it started to be used instead of writing a new parsers for these particular use cases.
About the naming, I would like to avoid using more aggressive names as non_compliant.

Reply:
while I would like more of a scarlet colour on this "red flag", I guess that will have to be enough :smile: